pkgs/development/libraries/zlib/default.nix

```
@@ -104,7 +104,8 @@ stdenv.mkDerivation (finalAttrs: {
     # Non-typical naming confuses libtool which then refuses to use zlib's DLL
     # in some cases, e.g. when compiling libpng.
     + lib.optionalString (stdenv.hostPlatform.isMinGW && shared) ''
-      ln -s zlib1.dll $out/bin/libz.dll
+      mv $out/{lib,bin}
```

I think this is too aggressive. Probably needs to be something like:

```suggestion
      mkdir -p $out/bin
      mv $out/lib/*.dll* $out/bin/
```

Even that is too aggressive for a generic, as the convention is typically to have the DLL file under `/bin` and the `.dll.a` file under `/lib`, but I don't see a `.dll.a` file with zlib, so the above is probably fine for this use case.